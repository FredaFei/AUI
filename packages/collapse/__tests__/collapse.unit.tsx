import * as renderer from 'react-test-renderer'
import React from 'react'
import { mount} from 'enzyme'

import Collapse from '../collapse'
import Pane from "../pane";

describe('测试 Scroll 以下行为', () => {
  it('Collapse 存在', () => {
    const json = renderer.create(<Collapse/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('Collapse 设置 defaultActiveKey', () => {
    const wrapper = mount(<Collapse defaultActiveKey={['1']}>
      <Pane header="This is panel header 1" name="1">
        <p>This is pane content 1</p>
      </Pane>
      <Pane header="This is panel header 2" name="2">
        <p>This is pane content 2</p>
      </Pane>
      <Pane header="This is panel header 3" name="3">
        <p>This is pane content 3</p>
      </Pane>
    </Collapse>)

    expect(wrapper.props().defaultActiveKey.length).toEqual(1)
    expect(wrapper.props().defaultActiveKey['0']).toEqual('1')
  })

})