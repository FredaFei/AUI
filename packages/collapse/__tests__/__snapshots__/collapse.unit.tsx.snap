// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Collapse 以下行为 Collapse 存在 1`] = `
<div
  className="am-collapse-wrapper "
  data-role="Collapse"
/>
`;
